app/src/main/java/com/gatheringhallstudios/mhworlddatabase/features/weapons/detail/WeaponDetailFragment.kt

```diff
@@ -285,9 +289,55 @@ class WeaponDetailFragment : Fragment() {
         view.findViewById<TextView>(R.id.shelling_level_value).text = getString(R.string.skill_level_short_qty, weapon.shelling_level)
     }
 
-    private fun bindHuntingHorn(weapon: Weapon, view: View) {
+    private fun bindHuntingHorn(weapon: Weapon, melodies: List<WeaponMelody>?, view: View) {
         populateSharpness(weapon.sharpnessData, view)
-        view.findViewById<TextView>(R.id.notes_value).text = weapon.notes.toString()
+        notes_layout.removeAllViews()
+        weapon.notes!!.forEach {
+            val noteIcon = ImageView(context)
+            noteIcon.layoutParams = ViewGroup.LayoutParams(resources.getDimension(R.dimen.image_size_small).toInt(), resources.getDimension(R.dimen.image_size_small).toInt())
+            when(it) {
+                'W' -> noteIcon.setImageDrawable(loadIconFor(NoteType.WHITE))
+                'R' -> noteIcon.setImageDrawable(loadIconFor(NoteType.RED))
+                'B' -> noteIcon.setImageDrawable(loadIconFor(NoteType.BLUE))
+                'O' -> noteIcon.setImageDrawable(loadIconFor(NoteType.ORANGE))
+                'Y' -> noteIcon.setImageDrawable(loadIconFor(NoteType.YELLOW))
+                'P' -> noteIcon.setImageDrawable(loadIconFor(NoteType.PURPLE))
+                'G' -> noteIcon.setImageDrawable(loadIconFor(NoteType.GREEN))
+                'C' -> noteIcon.setImageDrawable(loadIconFor(NoteType.CYAN))
+            }
```

Comment:
not gonna decline for this, but you can do `val image = when (it) {` where each case resolves to `loadIconFor(NoteType.Whatever)`, and then `noteIcon.setImageDrawable(image)`. At this point its whatever but just letting you know that kotlin when statements can return results.